epic(eslint-config): rework and restructure eslint-config-esl module (#2968)
exadel-inc/esl: packages/eslint-config/README.md

Once installed, the configuration needs to be added in eslint configuration file:

```js
// Import only configs you need
import {lang, base, codestyle, medium, strict} from '@exadel/eslint-config-esl';

export default [
    // Default lang (parsers) configs, if required
    //...lang.js,
    //...lang.ts,
        
    // ESL ESLint configuration of your choice
    ...strict // or medium, or base, or codestyle   
    // Note: recomended sets of eslint / typscript-esint are included in base, medium and strict config.     
        
    // ESLint user configuration ...
];
```

> Peaase note '@exadel/eslint-config-esl' is ESM only package.

Reviewer: ```suggestion
Please note '@exadel/eslint-config-esl' is an ESM-only package.
```

You either need to use ESM in your project or run eslint independently with `.mjs` configuration file.

<a name="configuration"></a>

### Configuration

> ESL Shared ESLint config splitted into several configurations/bunches:

Reviewer 2: ```suggestion
ESL Shared ESLint Configuration is split into several configurations/bunches:
```

- `base` - basic light linting configuration, uses `@eslint/js/` recommended and `typescript-eslint` recommended ruleset as a base. 

> - `codestyle` - code style configuration, additionally uses `@stylistic` and `import` plugins. Does not includes `base` rules.

Reviewer: ```suggestion
- `codestyle` - code style configuration, additionally uses `@stylistic` and `import` plugins. Does not include `base` rules.
```

- `medium` - medium configuration, combination `base` and `codestyle` rules.
- `strict` - strict configuration, includes `base` and `codestyle`, extended with `sonarjs` and `tsdoc` plugins and the most restrictive settings.

> The `base` ruleset recomended for consumers to have basic code checks similar to ESL source code.

Reviewer: ```suggestion
The `base` ruleset recommended for consumers to have basic code checks similar to ESL source code.
```

The `codestyle` ruleset is recommended for projects that want to follow ESL code style.
The `medium` ruleset is a shortcut for projects that want to have both basic and code style checks.

> The `strict` ruleset is recommended for projects that commpletely follow ESL code style and want to have the most restrictive checks.

Reviewer: ```suggestion
The `strict` ruleset is recommended for projects that completely follow ESL code style and want to have the most restrictive checks.
```

        ecmaVersion: 2017,
        sourceType: 'module',
        parserOptions: {
            projectService: true
        },
        globals: {
            ...globals.browser
        }
    }
};
```

### Inner Plugins

ESL Shared ESLint Configuration includes several inner plugins that are used to provide additional rules and configurations for ESLint.
Here is the list of included plugins and their ESLint aliases:

- `@eslint/js` - basic ESLint rules from [eslint](https://eslint.org/) project.
- `typescript-eslint` - TypeScript specific rules from [TypeScript ESLint](https://typescript-eslint.io/) project.
- `@stylistic` - code style rules from [@stylistic](https://eslint.style/) project.
- `import` - rules for imports from [eslint-plugin-import-x](https://www.npmjs.com/package/eslint-plugin-import-x);
- `tsdoc` - rules for TSDoc comments from [eslint-plugin-tsdoc](https://www.npmjs.com/package/eslint-plugin-tsdoc);
- `sonarjs` - rules for code quality from [eslint-plugin-sonarjs](https://www.npmjs.com/package/eslint-plugin-sonarjs);
- `editorconfig` - rules for EditorConfig from [eslint-plugin-editorconfig](https://www.npmjs.com/package/eslint-plugin-editorconfig);

> All mentioned plugins are direct dependencies of `@exadel/eslint-config-esl` package you don't need to install them separately.

Reviewer: ```suggestion
All mentioned plugins are direct dependencies of `@exadel/eslint-config-esl` package, you don't need to install them separately.
```
